### ISSUE TITLE
More builtin compile-time filters: json, yaml, and jq

### ISSUE THREAD

**Issue by agentzh** [orinc/testml-fan#21]
These compile-time filters can be very handy, or we can just allow the TestML implementation allows plugins which implement custom compile-time filters.

**Comment by ingydotnet** [comment]
Yeah, maybe *data* filters should be compile time, and *code* (prologue) filters are runtime.

Or maybe for data filters we look for a compile time one first, then fall back to runtime.

Need to think on it more...

**Comment by agentzh** [comment]
@ingydotnet Yeah, that makes sense.
